### PR TITLE
Add GcpCallInvokerPool

### PR DESCRIPTION
**Pull request by evildour**
(This has been ported and cleaned up from https://github.com/GoogleCloudPlatform/google-cloud-dotnet/pull/2454)

### CODE REVIEW

**Comment by jskeet** [comment]
Thanks for this - will look through it carefully, and we'll need to work out how the GCP dependency fits in here. (It may well be fine to keep it as a dependency which is sometimes irrelevant, just for the sake of simplicity.)

**Comment by jskeet** [comment]
Okay, Chris and I have talked, and we *think* we'd want this in a new package (Google.Gax.Grpc.Gcp probably) to avoid this more general package from taking a GCP-specific dependency. (It's fine for Google.Gax.Grpc.Gcp to depend on Google.Gax.Grpc.) That would also mean we could merge this and even create a beta release of the new package, if that's helpful for the Bigtable work.

However, we want to discuss with with colleagues from other languages, to have a consistent approach. I've put it in an agenda item for Tuesday. Is that soon enough, or is this super-urgent?

File: src/Google.Api.Gax.Grpc/EqualityHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Google.Api.Gax.Grpc
+{
+    // TODO: These are copied from Firestore utilities. If we make them public, they can be reused there.
+    //       Just make sure we have everything Firestore needs here if we do decide to do that.
```
[the remark this answers is not in the document]
Comment (evildour) [reply src/Google.Api.Gax.Grpc/EqualityHelpers.cs]: Ok, I'll pull in anything from Firestore I didn't get already and update the namespace. Does such a namespace exist already? If not, is `Google.Api.Gax.Grpc.Utilities` a good choice?

[the remark this answers is not in the document]
Comment (jskeet) [reply src/Google.Api.Gax.Grpc/EqualityHelpers.cs]: If we want to make it public, I'd put it in Google.Api.Gax itself (possibly in Google.Api.Gax.Collections?). If it's staying internal to the same package as the rest of the code, the namespace doesn't matter.

**Comment by evildour** [comment]
That should be fine. I don't think this is super-urgent, but I'll double check internally.

File: src/Google.Api.Gax.Grpc/EqualityHelpers.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (evildour) [reply src/Google.Api.Gax.Grpc/EqualityHelpers.cs]: Ok sounds good. I'll wait until Tuesday though before moving anything.

**Comment by jskeet** [comment]
Closing as #274 does this instead.

**Pull request closed without merging**